velox/docs/functions/json.rst

```diff
@@ -2,6 +2,89 @@
 JSON Functions
 ==============
 
+JSON Format
+-----------
+JSON is a language-independent data format that represents data as
+human-readable text. A JSON text can represent a number, a boolean, a
+string, an array, an object, or a null, with slightly different grammar.
+For instance, a JSON text representing a string must escape all characters
+and enclose the string in double quotes, such as ``"123\n"``, whereas a JSON
+text representing a number does not need to, such as ``123``. A JSON text
+representing an array must enclose the array elements in square brackets,
+such as ``[1,2,3]``. More detailed grammar can be found in
+`this JSON introduction`_.
+
+.. _this JSON introduction: https://www.json.org
+
+Cast to JSON
+------------
+Casting a value from a supported type to JSON returns a JSON text that
+represents this value. Casting from BOOLEAN, TINYINT, SMALLINT, INTEGER,
+BIGINT, REAL, DOUBLE, DATE, TIMESTAMP, or VARCHAR is supported. Casting
+from ARRAY or ROW is supported when the element type of the array is one
+of the supported types, or when every field type of the row is one of the
+supported types. Casting from MAP is supported when the key type of the map
+is BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE, or VARCHAR
+and value type of the map is one of the supported types. Additionally,
+element types of ARRAY and MAP and field types of ROW are also allowed to
+be JSON. Behaviors of the casts are shown with the examples below:
+
+``SELECT CAST(NULL AS JSON); -- NULL``
+``SELECT CAST(1 AS JSON); -- JSON '1'``
+``SELECT CAST(9223372036854775807 AS JSON); -- JSON '9223372036854775807'``
+``SELECT CAST('abc' AS JSON); -- JSON '"abc"'``
+``SELECT CAST(true AS JSON); -- JSON 'true'``
+``SELECT CAST(1.234 AS JSON); -- JSON '1.234'``
+``SELECT CAST(ARRAY[1, 23, 456] AS JSON); -- JSON '[1,23,456]'``
+``SELECT CAST(ARRAY[1, NULL, 456] AS JSON); -- JSON '[1,null,456]'``
+``SELECT CAST(ARRAY[ARRAY[1, 23], ARRAY[456]] AS JSON); -- JSON '[[1,23],[456]]'``
+``SELECT CAST(MAP_FROM_ENTRIES(ARRAY[('k1', 1), ('k2', 23), ('k3', 456)]) AS JSON); -- JSON '{"k1":1,"k2":23,"k3":456}'``
+``SELECT CAST(CAST(ROW(123, 'abc', true, JSON '["a"]') AS ROW(v1 BIGINT, v2 VARCHAR, v3 BOOLEAN, v4 JSON)) AS JSON); -- JSON '[123,"abc",true,["a"]]'``
+
+Notice that casting from NULL to JSON is not straightforward. Casting from
+a standalone NULL will produce a SQL NULL instead of JSON 'null'. However,
+when casting from arrays or map containing NULLs, the produced JSON will
+have nulls in it.
+
+Another thing to be aware of is that when casting from ROW to JSON, the
+result is a JSON array rather than a JSON object. This is because positions
```

Comment:
This is outside the scope of this PR, but Array is an interesting choice. In the ROW example above, I see that we do not save the field names in the JSON output. I feel field names are equally important as positions. Should the output be Array of objects? I also think JSON array elements should be of the same type. Is the above output accurate?

Reply:
Hi Deepak, this is an interesting idea! In fact, this paragraph is copied from Presto documentation (https://prestodb.io/docs/current/functions/json.html, second note under "Cast to JSON"). 

JSON allows elements in an array to be of different types. I tested the example of casting-row-to-json above with java-Presto and it is correct.